content/engine/install/debian.md

```diff
@@ -59,7 +59,7 @@ conflicts with the versions bundled with Docker Engine.
 Run the following command to uninstall all conflicting packages:
 
 ```console
-$ for pkg in docker.io docker-doc docker-compose podman-docker containerd runc; do sudo apt-get remove $pkg; done
+$ sudo apt-get remove docker.io docker-doc docker-compose podman-docker containerd runc
```

Comment:
Using a loop was an intentional change; without a loop, this command may fail, and some packages still installed;  see

- https://github.com/docker/docs/pull/17415
- https://github.com/docker/docs/issues/17405


> resulting in having to confirm every uninstallation.

Yeah, that's rather inconvenient, and II can see that being inconvenient. Not sure what a better solutions would be for that, without this step becoming overly complex. We could detect which packages are installed, but probably that would make it either an extra step, or "complex",

If your situation allows it, and needs to be run non-interactively, it's of course use your own variant 😅

Reply:
I've just looked to these issues, and tested a few package removal combinations. Turns out, apt-get only fails if a package does not exist. That is, when there is no such package in the package repository. When a *real* package is not installed, apt continues normally:

```
$ sudo apt-get remove docker.io podman-docker
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package 'podman-docker' is not installed, so not removed
...
The following packages will be REMOVED:
  docker.io
0 upgraded, 0 newly installed, 1 to remove and 0 not upgraded.
After this operation, 151 MB disk space will be freed.
Do you want to continue? [Y/n] 
```

According to docs itself, these guides are for debian 11 or higher, ubuntu 20.04 or higher, and their derivatives. All of the packages in the changed lines are in the repositories already. So, I think my change would not break something, or would it?

In #17405, the problem was related with nonexistent docker-engine package. Looks like that package is already removed from here.

Reply:
I'm not sure when the `docker-engine` package was removed from here but indeed, `apt-get` will only skip and not error out if the package exists but was not installed. Given that we don't list any nonexistent packages here, I think we should be safe to remove the for-loop now. WDYT @thaJeztah?

Besides, it's been a pretty long time since we shipped packages under those nonexistent (e.g. `docker-engine`) names, no? We're only backwards compatible with oldstable, so this should be fine I think.